lib/nandi/multi_database.rb

```diff
@@ -0,0 +1,157 @@
+      # @return [Integer]
+      attr_accessor :access_exclusive_lock_timeout
+
+      # The default statement timeout for migrations that take ACCESS EXCLUSIVE
+      # locks. Can be overridden by way of the `set_statement_timeout` class
+      # method in a given migration. Default: 1500ms.
+      # @return [Integer]
+      attr_accessor :access_exclusive_statement_timeout
+
+      # The maximum lock timeout for migrations that take an ACCESS EXCLUSIVE
+      # lock and therefore block all reads and writes. Default: 5,000ms.
+      # @return [Integer]
+      attr_accessor :access_exclusive_statement_timeout_limit
+
+      # The maximum statement timeout for migrations that take an ACCESS
+      # EXCLUSIVE lock and therefore block all reads and writes. Default: 1500ms.
+      # @return [Integer]
+      attr_accessor :access_exclusive_lock_timeout_limit
+
+      # The minimum statement timeout for migrations that take place concurrently.
+      # Default: 3,600,000ms (ie, 3 hours).
+      # @return [Integer]
+      attr_accessor :concurrent_statement_timeout_limit
+
+      # The minimum lock timeout for migrations that take place concurrently.
+      # Default: 3,600,000ms (ie, 3 hours).
+      # @return [Integer]
+      attr_accessor :concurrent_lock_timeout_limit
+
+      # The directory for output files. Default: `db/migrate`
+      # @return [String]
+      attr_accessor :output_directory
+
+      attr_reader :name, :default
+
+      attr_accessor :migration_directory,
+                    :lockfile_name
+
+      def initialize(name:, config:)
+        @name = name
+        @default = (true if @name == :primary) || config[:default]
+
+        # Paths and files
+        @migration_directory = config[:migration_directory] || "db/#{prefixed_path(name, default)}safe_migrations"
+        @output_directory = config[:output_directory] || "db/#{prefixed_path(name, default)}migrate"
+        @lockfile_name = config[:lockfile_name] || ".#{prefixed_path(name, default)}nandilock.yml"
+
+        # Timeout limits
+        @access_exclusive_lock_timeout = config[:access_exclusive_lock_timeout] || DEFAULT_ACCESS_EXCLUSIVE_LOCK_TIMEOUT
+        @access_exclusive_statement_timeout = config[:access_exclusive_statement_timeout] || DEFAULT_ACCESS_EXCLUSIVE_STATEMENT_TIMEOUT
+        @access_exclusive_lock_timeout_limit = config[:access_exclusive_lock_timeout_limit] || DEFAULT_ACCESS_EXCLUSIVE_LOCK_TIMEOUT_LIMIT
+        @access_exclusive_statement_timeout_limit = config[:access_exclusive_statement_timeout_limit] || DEFAULT_ACCESS_EXCLUSIVE_STATEMENT_TIMEOUT_LIMIT
+        @concurrent_lock_timeout_limit = config[:concurrent_lock_timeout_limit] || DEFAULT_CONCURRENT_TIMEOUT_LIMIT
+        @concurrent_statement_timeout_limit = config[:concurrent_statement_timeout_limit] || DEFAULT_CONCURRENT_TIMEOUT_LIMIT
+      end
+
+      private
+
+      def prefixed_path(name, default)
+        name.nil? || default ? "" : "#{name}_"
```

Comment:
```suggestion
        default ? "" : "#{name}_"
```
should it be this? Name can only be missing for single-db configs